api/metrics/v1/rules.go

```diff
@@ -71,26 +89,32 @@ func (rh *rulesHandler) get(w http.ResponseWriter, r *http.Request) {
 			switch r := rawRules.Groups[i].Rules[j].(type) {
 			case rules.RecordingRule:
 				if r.Labels.AdditionalProperties == nil {
-					r.Labels.AdditionalProperties = make(map[string]string)
+					http.Error(w, "rule has no label", http.StatusInternalServerError)
```

Comment:
I'm on the fence too about this. 
But it is kind of a fault on our end, as this switch case is only entered when there is a rule in a rule group. 
So we have a rule, but with no label which is saved in rules storage, which means something is very wrong, as we do not allow rules without tenant labels to be saved in the first place. So a `5xx` status code would seem apt, right?

Reply:
i kind of think that we should continue to enforce the labels on the read path rather than return a non-200 error code. If the rules were loaded into the rules backend some other way than through the observatorium API, then this current changeset would make the rules un-fetchable. What's the downside to enforcing again and returning 200s?

Reply:
The downside is that the original problem mentioned in the ticket would persist.

If we add rules to the rules backend some other way and don't add in the tenant labels, the rules would again be propagated and configured on the Ruler and the series generated won't have tenant labels on them as well. So, the tenant the rules belong to can't query them.
But if we enforce it on the read path, then in the above condition the observatorium API will return a 200 status code and the rules with added tenant labels, even though the series generated from rules aren't actually being queryable (due to missing labels).

This changeset ensures that all rules in the rules backend have a tenant label. So even if we load the rules some other way in the rules backend, we must provide the tenant label. 🙂

Reply:
@saswatamcode i think you perhaps misunderstood my comment. I'm proposing we enforce labels on both the PUT and the GET (for a single tenant). As far as I can tell, there is no downside at all to this and it protects us in the event that rules are sideloaded into the ruler.

Reply:
Ack! Yes, there's no downside then. Let me push a commit!

Reply:
Ok nice :) one last thing, it looks like both the PUT and the GET handlers use the exact same logic to enforce labels. Maybe this whole code block could be abstracted to a func that takes the tenant name and rules pre-typeswitch and applies the labels. WDYT?

Reply:
Yes, that would be better!